cluster-api-azure-controller.yaml

```diff
@@ -0,0 +1,59 @@
+package:
+  name: cluster-api-azure-controller
+  version: "1.20.2"
+  epoch: 0
+  description: Cluster API implementation for Microsoft Azure
+  copyright:
+    - license: Apache-2.0
+  dependencies:
+    runtime:
+      - ca-certificates-bundle
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/kubernetes-sigs/cluster-api-provider-azure
+      tag: v${{package.version}}
+      expected-commit: 2a4b30db466c22b80f77e5c704459b197412ee45
+
+  - uses: go/build
+    with:
+      packages: .
+      output: cluster-api-azure-controller
+      ldflags: |
+        -buildid= ""
+        -X sigs.k8s.io/cluster-api-provider-azure/version.buildDate=$(date -u -d "@$SOURCE_DATE_EPOCH" +'%Y-%m-%dT%H:%M:%SZ')
+        -X sigs.k8s.io/cluster-api-provider-azure/version.gitCommit=$(git rev-parse HEAD)
+        -X sigs.k8s.io/cluster-api-provider-azure/version.gitVersion=$(git describe --dirty --tags --always)
+        -X sigs.k8s.io/cluster-api-provider-azure/version.gitTreeState=$(shell git diff --quiet >/dev/null 2>&1 || echo "dirty" || echo "clean")
```

Comment:
```suggestion
        -X sigs.k8s.io/cluster-api-provider-azure/version.gitTreeState=$(git diff --quiet >/dev/null 2>&1 || echo "dirty" || echo "clean")
```

Reply:
This only works with `shell`. Tested it. Not changing current code.